envoy/http/persistent_quic_info.h

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include <memory>
+
+namespace Envoy {
+namespace Http {
+
+// Store quic helpers which can be shared between connections and must live beyond the lifetime of
+// individual connections. When used in HTTP/3 upstream, it should be owned by cluster and shared
+// across its HTTP/3 connection pools. This an opaque placeholder is needed so that an
+// implementation can be passed around while the QUICHE members which are behind ENVOY_ENABLE_QUIC
+// preprocessor in the actual implementation can be hidden from the Envoy intefaces.
+struct PersistentQuicInfo {
```

Comment:
Awesome comment thank you.

I did ask in a prior version if we could try a version where we include the accessors we want with proper Envoy style, and define guard them here, rather than having this always empty and dynamic casting later on.  Did you try that and it was uglier or did it just get missed?

Reply:
Yes, see comment https://github.com/envoyproxy/envoy/pull/19862#discussion_r819812811. TLDR is there would require much more refectory to untangle a circular dependency to achieve that.

Reply:
SG, as long as we tried :-)